### ISSUE TITLE
[SAE] ManifoldSAE.forward() reports amplitudes that are not the reconstruction code for TopK/JumpReLU

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#610]
**Verified — CONFIRMED.**

`forward` computes (`gamfit/torch/manifold_sae.py:706-708`):
```python
amp = F.softplus(amp_logits)
assignments, gate_pre = self.sparsity(amp_logits)
z = self.sparsity.compose_code(assignments, amp)
```
but `compose_code` returns just `assignments` (ignoring `amp`) for `softmax_topk` and `jumprelu` (~546-559), while the output stores `amplitudes=amp` (~719). Since softplus is always positive, atoms that contribute **zero** to reconstruction (z=0, outside top-k / below threshold) still report a positive `amplitudes`. Downstream interpretability code reading `amplitudes` will conclude an atom was active when it did not contribute.

**Fix:** expose the actual reconstruction code as `z` and report `amplitudes = z` (or rename the softplus tensor to `raw_magnitudes`); ideally return separate `raw_logits` / `magnitudes` / `gates` / `codes` fields.

_Source: static SAE/manifold-SAE audit; verified against current code at commit 3c1a4efa2._

**Comment by SauersML** [comment]
Fixed on main in `7ae57a63a`: forward now reports `amplitudes=z` (the actual reconstruction code) with the pre-mask softplus exposed separately as `raw_magnitudes` (manifold_sae.py:804-810).
